learning-development/r.qmd

```diff
@@ -389,6 +389,170 @@ To round numbers upwards in a way that is consistent with SQL and other programm
+# Shiny app ===================================================================
+# Minimal example code for a shiny app with interactive map
+# Can copy and paste into a single app.R file and run to test
+
+# Pre-processing of data (do this in a separate script to the app) ============
+# Load dependencies -----------------------------------------------------------
+library(leaflet) # using leaflet for the interactive map
+library(dplyr) # used for data processing
+library(sf) # used for reading geospatial data
+
+# Pre-process data before app -------------------------------------------------
+# Create an example file using the locations look ups
+example_data <- read.csv("https://raw.githubusercontent.com/dfe-analytical-services/dfe-published-data-qa/main/data/lsips.csv") %>% 
+  mutate(example_count = sample(38)) # number of rows in data frame
+
+# Pull in boundary data from the Open Geography Portal
+# Download GeoPackage format and save in repo from:
+# https://geoportal.statistics.gov.uk/datasets/72bd510b46a7483194237ede9db0a267_0/explore
+boundary_data <- st_read("Local_Skills_Improvement_Plan_Areas_August_2023_Boundaries_EN_BUC_3056517900557747412.gpkg")
+
+# Join the data together and save ready for the app
+map_data <- boundary_data %>% 
+  right_join(example_data, by = join_by(LSIP23CD == lsip_code)) %>%
+  # Convert coordinates to World Geodetic System 
+  st_transform(crs = 4326)
+
+saveRDS(map_data, "map_data.RDS")
+
+# R Shiny App =================================================================
+# Global.R file ---------------------------------------------------------------
+library(shiny)
+library(leaflet)
+
+# Load in map data
+map_data <- readRDS("map_data.RDS")
+
+# ui.R file -------------------------------------------------------------------
+ui <- fluidPage(
+  # Show interactive map
+  leafletOutput("example_map")
+)
+
+# server.R file ---------------------------------------------------------------
+server <- function(input, output) {
+
+    output$example_map <- renderLeaflet({
+      # Set the color scheme and scale
+      pal_fun <- colorNumeric(
+        "Blues", 
+        domain = c(
+          min(map_data$example_count), 
+          max(map_data$example_count)
+          )
+        )
+      
+      # Set a pop up
+      map_popup <- paste(
+        map_data$example_count, 
+        " example count for ", 
+        map_data$lsip_name
+        )
+      
+      # Create the map 
+      map <- leaflet(
+        map_data, 
+        # Take off annoying scrolling, personal preference
+        options = leafletOptions(scrollWheelZoom = FALSE)
+        ) %>%
+        # Set the basemap (this is a good neutral one)
+        addProviderTiles(providers$CartoDB.PositronNoLabels) %>%
+        # Add the shaded regions
+        addPolygons(
+          color = "black",
+          weight = 1,
+          fillColor = pal_fun(map_data[["example_count"]]),
+          popup = map_popup
+        ) %>%
+        # Add a legend to the map
```

Comment:
I'm guessing this does not include the fix to the legend we need for accessibility 😄

Reply:
You are correct, it does not 😅